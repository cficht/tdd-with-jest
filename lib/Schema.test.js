const { Schema } = require('../lib/Schema.js');

describe('Schema testing', () => {

  const schema = new Schema({
    name: {
      type: String,
      required: true
    },
    age: {
      type: Number,
      required: true
    },
    weight: {
      type: String
    }
  });

  const spot = {
    name: 'spot',
    age: 5,
    weight: '20 lbs'
  };

  const rover = {
    name: 'rover',
    age: '10'
  };

  const who = {
    age: 'hi'
  };

  const otherdog = {
    name: 'brian',
    weight: '50 lbs'
  };

  it('Returns spot as is', () => {
    expect(schema.validate(spot)).toEqual(spot);
  });

  it('Returns rover with the age turned into a number', () => {
    expect(schema.validate(rover)).toEqual({
      name: 'rover',
      age: 10
    });
  });

  it('Throws an error for an object missing name', () => {
    expect(() => schema.validate(who)).toThrowErrorMatchingSnapshot();
  });

  it('Throws an error for an object missing age', () => {
    expect(() => schema.validate(otherdog)).toThrowErrorMatchingSnapshot();
  });
});
