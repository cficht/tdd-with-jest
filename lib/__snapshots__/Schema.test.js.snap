// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema testing Throws an error for an object missing age 1`] = `"undefined is required"`;

exports[`Schema testing Throws an error for an object missing name 1`] = `"undefined is required"`;
